fem/tmop.cpp

```diff
@@ -3014,31 +3039,51 @@ void TMOP_Integrator::EnableSurfaceFittingFromSource(
 }
 #endif
 
-void TMOP_Integrator::GetSurfaceFittingErrors(double &err_avg, double &err_max)
+void TMOP_Integrator::GetSurfaceFittingErrors(const Vector &pos,
+                                              double &err_avg, double &err_max)
 {
-   MFEM_VERIFY(surf_fit_gf, "Surface fitting has not been enabled.");
+   MFEM_VERIFY(surf_fit_marker, "Surface fitting has not been enabled.");
 
+   const FiniteElementSpace *fes =
+      (surf_fit_gf) ? surf_fit_gf->FESpace() : surf_fit_pos->FESpace();
 #ifdef MFEM_USE_MPI
-   auto pfes =
-      dynamic_cast<const ParFiniteElementSpace *>(surf_fit_gf->FESpace());
+   auto pfes = dynamic_cast<const ParFiniteElementSpace *>(fes);
    bool parallel = (pfes) ? true : false;
 #endif
 
+   int dim = fes->GetMesh()->Dimension();
+   if (surf_fit_pos)
+   { MFEM_VERIFY(fes->GetOrdering() == Ordering::byNODES, "Not supported."); }
```

Comment:
I think we should support ordering `byVDIM` since some of TMOP users rely on it. We could instead enforce a softer constraint that the ordering of `surf_fit_pos` and mesh nodes is the same.

Reply:
done.